pkg/pillar/cmd/zedrouter/zedrouter.go

```diff
@@ -2647,3 +2663,167 @@ func handleDNSDelete(ctxArg interface{}, key string,
+			match, err := networkInstancesWithUplinkOverlap(ctx, network0, network1)
+			if err != nil {
+				return false, err
+			}
+			if match {
+				if matchACLsForPortMap(ulCfg0.ACLs, ulCfg1.ACLs) {
+					return true, nil
+				}
+			}
+		}
+	}
+	return false, nil
+}
+
+// network instance or, sharing common uplink (ephimeral or, physical port)
+func networkInstancesWithUplinkOverlap(ctx *zedrouterContext, network string,
+	network1 string) (bool, error) {
+	if network == network1 {
+		log.Debugf("Same Network Instance, %s\n", network)
+		return true, nil
+	}
+	netInstConfig := lookupNetworkInstanceConfig(ctx, network)
+	netInstConfig1 := lookupNetworkInstanceConfig(ctx, network1)
+	if netInstConfig == nil || netInstConfig1 == nil {
+		return false, errors.New("invalid network instance")
+	}
+	portName := netInstConfig.Port
+	portName1 := netInstConfig1.Port
+	if portName == "" || portName1 == "" {
+		return false, nil
+	}
+	if portName == portName1 {
+		return true, nil
+	}
+	// of uplink type
+	if portName == "uplink" {
+		return isUpLink(ctx, portName1)
+	}
+	if portName1 == "uplink" {
+		return isUpLink(ctx, portName)
+	}
+
+	// of free uplink Type
+	// for free uplink type, match uplink also
+	// as free uplink group is a subset of uplink
+	if portName == "freeuplink" {
+		if portName1 == "uplink" {
+			return true, nil
+		}
+		return isFreeUpLink(ctx, portName1)
+	}
+	if portName1 == "freeuplink" {
+		if portName == "uplink" {
+			return true, nil
+		}
+		return isFreeUpLink(ctx, portName)
+	}
+	return false, nil
+}
+
+func isFreeUpLink(ctx *zedrouterContext, portName string) (bool, error) {
+	if portName == "freeuplink" {
+		return true, nil
+	}
+	if !types.IsPort(*ctx.deviceNetworkStatus, portName) {
+		return false, errors.New("invalid port")
+	}
+	// check whether this physical port belongs to freeUpLink group
+	return types.IsFreeMgmtPort(*ctx.deviceNetworkStatus, portName), nil
+}
+
+func isUpLink(ctx *zedrouterContext, portName string) (bool, error) {
+	if portName == "uplink" {
+		return true, nil
+	}
+	if !types.IsPort(*ctx.deviceNetworkStatus, portName) {
+		return false, errors.New("invalid port")
+	}
+	// check whether this physical port belongs to UpLink group
+	return types.IsMgmtPort(*ctx.deviceNetworkStatus, portName), nil
+}
+
+func matchACLsForPortMap(ACLs0 []types.ACE, ACLs1 []types.ACE) bool {
+	for _, ace0 := range ACLs0 {
+		for _, action0 := range ace0.Actions {
+			// not a portmap rule
+			if !action0.PortMap {
+				continue
+			}
+			for _, ace1 := range ACLs1 {
+				for _, action1 := range ace1.Actions {
+					// not a portmap rule
+					if !action1.PortMap {
+						continue
+					}
+					// for target port
+					if action0.TargetPort == action1.TargetPort {
+						return true
+					}
+					// for ingress port
+					if checkForMatchCondition(ace0, ace1) {
+						return true
+					}
+				}
+			}
+		}
+	}
+	return false
+}
+
+func checkForMatchCondition(ace0 types.ACE, ace1 types.ACE) bool {
```

Comment:
This doesn't look like.
If match1 has protocol=tcp and match2 has protocol=tcp you conclude it is a conflict?
I think the simple check for portmap is (in addition to the IfNameList having overlap) is that the lport and protocol both match.

Reply:
sure, implemented.